react/features/prejoin/actions.js

```diff
@@ -554,3 +556,30 @@ function startConference() {
         type: PREJOIN_START_CONFERENCE
     };
 }
+
+/**
+ * Action used to mute/unmute prejoin video.
+ * It destroys/recreates prejoin video tracks.
+ *
+ * @param {boolean} shouldMute - If should mute or unmute.
+ * @returns {Function}
+ */
+export function mutePrejoinVideo(shouldMute: boolean) {
```

Comment:
I don't think this is the right approach here. I think you might have ended up in this situation because of a non-obvious design choice we made: base/media vs base/tracks. base/media expresses intent, and base/tracks reality. Here we are introducing a new track store, which is used before the conference starts, so this would be a replacement for base/tracks.

Thus, there is no need to have separate muted flags at all, you can use the ones in base/media, instead of duplicating them in the middleware.

Then, `isPrejoinVideoTrackSet` can disappear completely, because what you need to do is either destroy the track or create it if it's not there yet.

This means that `setPrejoinAudioMuted` and the video equivalent also have to go. Sorry I didn't catch that during the initial review.

Reply:
Makes sense. Updated accordingly.